stock_split_picking/models/stock_picking.py

```diff
@@ -71,7 +71,8 @@ def split_process(self):
                 new_moves.mapped("move_line_ids").write(
                     {"picking_id": backorder_picking.id}
                 )
-                new_moves._action_assign()
+                if backorder_picking.picking_type_id.reservation_method == "at_confirm":
```

Comment:
@AnizR I'm not sure either it's sufficent either it's not the good approach.

Looking at this, I think there should be additionnal filters:

https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_move.py#L1409

But, shouldn't `_action_confirm()`be called at this moment (and not above - line 64) ? So, the base code is called and you don't bother about reservation method or...

Reply:
In deed, moving the call to `_action_confirm` will do the trick! Thanks